src/zenml/zen_stores/secrets_stores/base_secrets_store.py

```diff
@@ -75,60 +106,102 @@ def __init__(
+            StoreEvent.USER_DELETED, self._on_user_deleted
+        )
+
         try:
             self._initialize()
         except Exception as e:
             raise RuntimeError(
                 f"Error initializing {self.type.value} secrets store: {str(e)}"
             ) from e
 
+    @staticmethod
+    def _load_external_store_class(
+        store_config: SecretsStoreConfiguration,
+    ) -> Type["BaseSecretsStore"]:
+        """Loads the external secrets store class from the given config.
+
+        Args:
+            store_config: The configuration of the secrets store.
+
+        Returns:
+            The secrets store class corresponding to the configured external
+            secrets store.
+
+        Raises:
+            ValueError: If the configured class path cannot be imported or is
+                not a subclass of `BaseSecretsStore`.
+        """
+        # Ensured through Pydantic root validation
+        assert store_config.class_path is not None
+
+        # Import the class dynamically
+        try:
+            store_class = import_class_by_path(store_config.class_path)
+        except (ImportError, AttributeError) as e:
+            raise ValueError(
+                f"Could not import class `{store_config.class_path}`: {str(e)}"
+            ) from e
+
+        if not issubclass(store_class, BaseSecretsStore):
+            raise ValueError(
+                f"Class `{store_config.class_path}` is not a subclass of "
+                f"`BaseSecretsStore`."
+            )
+
+        return store_class
+
     @staticmethod
     def get_store_class(
-        store_type: SecretsStoreType,
+        store_config: SecretsStoreConfiguration,
     ) -> Type["BaseSecretsStore"]:
         """Returns the class of the given secrets store type.
 
         Args:
-            store_type: The type of the secrets store to get the class for.
+            store_config: The configuration of the secrets store.
 
         Returns:
-            The class of the given store type or None if the type is unknown.
+            The class corresponding to the configured secrets store or None if
+            the type is unknown.
 
         Raises:
             TypeError: If the secrets store type is unsupported.
         """
-        if store_type == SecretsStoreType.SQL:
+        if store_config.type == SecretsStoreType.SQL:
             from zenml.zen_stores.secrets_stores.sql_secrets_store import (
                 SqlSecretsStore,
             )
 
             return SqlSecretsStore
-        elif store_type == SecretsStoreType.REST:
+
+        if store_config.type == SecretsStoreType.REST:
             from zenml.zen_stores.secrets_stores.rest_secrets_store import (
                 RestSecretsStore,
             )
 
             return RestSecretsStore
-        else:
+
+        if store_config.type == SecretsStoreType.AWS:
+            store_config.class_path = "zenml.zen_stores.secrets_stores.aws_secrets_store.AWSSecretsStore"
```

Comment:
Why is this handled differently than the SQL/Rest cast and the classes aren't simply imported here?

Reply:
Same as the root validator, a left-over from another time when secrets stores were located inside integrations. This can be unified now.